### ISSUE TITLE
Minor Questions/Remarks

### ISSUE THREAD

**Issue by araffin** [kachayev/gym-microrts-paper-sb3#2]
Hello,
nice project =).

Using dict observation is actually a nice trick to implement custom things without rewriting everything that I didn't think of ;).

You are also probably aware of the `MaskablePPO` implementation in SB3 contrib, but in case, here is the link: https://sb3-contrib.readthedocs.io/en/master/modules/ppo_mask.html

I also saw that: https://github.com/kachayev/gym-microrts-paper-sb3/commit/ff6b7d644b9a9e182684c8791c914983382589e4
what is the issue with more recent SB3 version?

**Comment by kachayev** [comment]
Hey! Thanks a lot for the feedback.

Yeah, I saw implementation from the contrib package. Though masking itself seemed to me like not the trickiest part of the implementation. And the very first idea with putting masks into observations dictionary to simply mask out logits in a custom `mlp_extractor` ... just worked out. That's why I didn't look deeply into `MaskablePPO` implementation. And because `PPO` itself in algorithms with such a large number of implementation details, the initial goal was to touch as little of the algo itself as possible (ideally, no changes at all).

Regarding the version, the latest version of `gym-microrts==0.5.0.` requires `gym>=0.20.0` which turned out to be incompatible with `stable_baselines3==1.3.0`. Previously I used MicroRTS 0.4.3 with SB3 1.3.0 but that required me to reimplement some pieces of the environment logic only available in the latest release. I'm not sure if Gym version is critical for MicroRTS, need to do more experiments with different version. It might be easier to wait for SB3 to add support for the latest Gym though 😀
